lib/hooks.dart

```diff
@@ -468,7 +449,7 @@ Ref<T> useRefInit<T>(T initialValue) => Ref.useRefInit(initialValue);
 /// ```
 ///
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#usememo>.
-T useMemo<T>(T Function() createFunction, [List<dynamic> dependencies]) =>
+T useMemo<T>(T Function() createFunction, [List<dynamic>? dependencies]) =>
```

Comment:
Same thing here - a call to `useMemo` with no dependencies... does nothing. Should dependencies default to `const []`?

Reply:
Yeah this one is a little weird; useMemo is documented as dependencies being required, but the TypeScript definition for it allows `undefined` for some reason, and that got (understandably) translated to an optional arg here.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/90a2a0544f5b47692e7782d7f5c5ebc36da3c0c5/types/react/index.d.ts#L1158-L1165

I think we could make a new, versioned API of this with required `dependencies`, but I don't think defaulting to `const []` in this implementation is a good idea, since it could result in unintentionally stale values if people forget the dependencies.

Also, the over_react analyzer plugin does warn when these aren't specified: https://github.com/Workiva/over_react/blob/8cd3f6cffadfac9bb703af666e904a6c9e965d2a/tools/analyzer_plugin/test/integration/diagnostics/exhaustive_deps_test_cases.dart#L1679

I don't feel super strongly about introducing a new `useMemo` just to make it required, and am fine with it being optional for now, so I'd vote to address that later down the road.